traveler-ios-ui/TravelerKitUI/ViewControllers/WishlistResultViewController.swift

```diff
@@ -0,0 +1,221 @@
+        let pagesNotRequested = pages.filter({ !wishlistResult.hasPage($0, pageSize: WishlistQuery.pageSize) })
+        for page in pagesNotRequested where !pagesLoading.contains(page) {
+            pagesLoading.insert(page)
+
+            let query = WishlistQuery(page: page, from: wishlistResult.fromDate, to: wishlistResult.toDate)
+            Traveler.fetchWishlist(query, identifier: page, delegate: self)
+        }
+    }
+}
+
+extension WishlistResultViewController: WishlistCellDelegate {
+    func wishlistCellDidPressRemoveButton(_ cell: WishlistCell) {
+        guard let indexPath = collectionView.indexPath(for: cell) else {
+            return
+        }
+
+        guard let item = wishlistResult?.items[indexPath.row] else {
+            return
+        }
+
+        let alert = UIAlertController(title: "Are you sure you want to remove this item?", message: nil, preferredStyle: .alert)
+        let noAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        let yesAction = UIAlertAction(title: "Yes", style: .default) { [unowned self] _ in
+            self.collectionView.performBatchUpdates({
+                if let result = Traveler.wishlistRemove([item], result: self.wishlistResult, delegate: self) {
+                    self.wishlistResult = result
+                    self._volatileResult = result
+
+                    self.collectionView.deleteItems(at: [indexPath])
+                }
+            }, completion: nil)
+        }
+
+        alert.addAction(noAction)
+        alert.addAction(yesAction)
+
+        present(alert, animated: true)
+    }
+}
+
+extension WishlistResultViewController: WishlistFetchDelegate {
+    public func wishlistFetchDidSucceedWith(_ result: WishlistResult, identifier: AnyHashable?) {
+        self.wishlistResult = _volatileResult
+
+        _ = (identifier as? Int).flatMap({ pagesLoading.remove($0) })
+
+        collectionView.reloadItems(at: collectionView.indexPathsForVisibleItems)
+    }
+
+    public func wishlistFetchDidFailWith(_ error: Error, identifier: AnyHashable?) {
+        _ = (identifier as? Int).flatMap({ pagesLoading.remove($0) })
+
+        // HANDLE CASE WHERE THE RESULTS DONT MATCH
+        switch error {
+        case WishlistResultError.resultMismatch:
+            delegate?.wishlistResultViewControllerDidRefresh(self)
+        default:
+            Log("Error fetching", data: error, level: .error)
+            break
+        }
+    }
+
+    public func previousResult() -> WishlistResult? {
+        return _volatileResult
+    }
+
+    public func wishlistFetchDidReceive(_ result: WishlistResult, identifier: AnyHashable?) {
+        _volatileResult = result
+    }
+}
+
+extension WishlistResultViewController: WishlistRemoveDelegate {
+    public func wishlistRemoveDidSucceed() {
+        // do nothing
+    }
+
+    public func wishlistRemoveDidFailWith(_ error: Error, result: WishlistResult?) {
+        self.wishlistResult = result
```

Comment:
might be good spot to guard if `result` is nil. and Log an error so developers know if they did something wrong